### PR TITLE
In the news notification UI

### PR DESCRIPTION
**Pull request by joewalsh**
First batch of changes for the in the news notification - working notification UI

### CODE REVIEW

**Comment by montehurd** [comment]
👍

**Comment by montehurd** [comment]
Travis may need to be restarted... I saw this in the log too...

`[x] /Users/travis/build/wikimedia/wikipedia-ios/WikipediaUnitTests/Code/NumberFormatterExtrasTests.swift:22:18: integer literal '312490123456' overflows when stored into 'Int'
[20:12:11]: ▸ number = 312490123456`

File: Wikipedia/Code/WMFNotificationsController.m [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+    UNUserNotificationCenter *center = [UNUserNotificationCenter currentNotificationCenter];
+    UNNotificationAction *readNowAction = [UNNotificationAction actionWithIdentifier:WMFInTheNewsNotificationReadNowActionIdentifier title:NSLocalizedString(@"in-the-news-notification-read-now-action-title", nil) options:UNNotificationActionOptionForeground];
+    UNNotificationAction *saveForLaterAction = [UNNotificationAction actionWithIdentifier:WMFInTheNewsNotificationShareActionIdentifier title:NSLocalizedString(@"in-the-news-notification-share-action-title", nil) options:UNNotificationActionOptionForeground];
+    UNNotificationAction *shareAction = [UNNotificationAction actionWithIdentifier:WMFInTheNewsNotificationSaveForLaterActionIdentifier title:NSLocalizedString(@"in-the-news-notification-save-for-later-action-title", nil) options:UNNotificationActionOptionForeground];
+
+    UNNotificationCategory *inTheNewsCategory = [UNNotificationCategory categoryWithIdentifier:WMFInTheNewsNotificationCategoryIdentifier actions:@[readNowAction, saveForLaterAction, shareAction] intentIdentifiers:@[] options:UNNotificationCategoryOptionNone];
```
Comment (montehurd) [inline Wikipedia/Code/WMFNotificationsController.m]: Seeing a crash adding nil items to this "actions" array...

Comment (joewalsh) [inline Wikipedia/Code/WMFNotificationsController.m]: @montehurd updated to nil check those items. Were you on a different locale? Did you see which was nil?

File: Wikipedia/Code/WMFArticleViewController.m [hunk 2]
```diff
@@ -1276,6 +1285,47 @@ - (void)shareAFactWithTextSnippet:(nullable NSString *)text {
     [self.shareOptionsController presentShareOptionsWithSnippet:text inViewController:self fromBarButtonItem:self.shareToolbarItem];
 }
 
+- (void)shareArticle {
```
Comment (coreyfloyd) [inline Wikipedia/Code/WMFArticleViewController.m]: Wait didn't we already have a share action?

**Comment by coreyfloyd** [comment]
@joewalsh looks good… just had question on the new share logic… looks like a lot was added… but I know we already had share actions so wondering if we moved this code from somewhere else or something.

**Comment by joewalsh** [comment]
@coreyfloyd weird - the diff shows more lines being added than actually were added...  I added shareArticle and the check for shouldShareArticle after article load.

**Comment by joewalsh** [comment]
@coreyfloyd it looks like that code was changed with monte's PR. will update this PR.

**Pull request merged**